flang/test/Lower/intrinsic-procedures/lbound.f90

```diff
@@ -2,69 +2,46 @@
 
 
 ! CHECK-LABEL: func @_QPlbound_test(
-! CHECK-SAME:  %[[VAL_0:.*]]: !fir.box<!fir.array<?x?xf32>>,
-! CHECK-SAME:  %[[VAL_1:.*]]: !fir.ref<i64>,
-! CHECK-SAME:  %[[VAL_2:.*]]: !fir.ref<i64>) {
+! CHECK:  %[[VAL_0:.*]] = fir.load %arg1 : !fir.ref<i64>
+! CHECK:  %[[VAL_1:.*]] = fir.address_of({{.*}})
 subroutine lbound_test(a, dim, res)
   real, dimension(:, :) :: a
   integer(8):: dim, res
-! CHECK:         %[[VAL_3:.*]] = arith.constant 1 : i64
-! CHECK:         fir.store %[[VAL_3]] to %[[VAL_2]] : !fir.ref<i64>
+! CHECK:         %[[VAL_2:.*]] = fir.convert %arg0 : (!fir.box<!fir.array<?x?xf32>>) -> !fir.box<none>
+! CHECK:         %[[VAL_3:.*]] = fir.convert %[[VAL_0]] : (i64) -> i32
+! CHECK:         %[[VAL_4:.*]] = fir.convert %[[VAL_1]] : (!fir.ref<!fir.char<1,{{.*}}>>) -> !fir.ref<i8>
+! CHECK:         %[[VAL_5:.*]] = fir.call @_FortranALboundDim(%[[VAL_2]], %[[VAL_3]], %[[VAL_4]], %{{.*}}) : (!fir.box<none>, i32, !fir.ref<i8>, i32) -> i64
+! CHECK:         fir.store %[[VAL_5]] to %arg2 : !fir.ref<i64>
   res = lbound(a, dim, 8)
 end subroutine
 
 ! CHECK-LABEL: func @_QPlbound_test_2(
-! CHECK-SAME:  %[[VAL_0:.*]]: !fir.box<!fir.array<?x?xf32>>,
-! CHECK-SAME:  %[[VAL_1:.*]]: !fir.ref<i64>,
-! CHECK-SAME:  %[[VAL_2:.*]]: !fir.ref<i64>) {
+! CHECK:  %[[VAL_0:.*]] = fir.load %arg1 : !fir.ref<i64>
+! CHECK:  %[[VAL_1:.*]] = fir.address_of
+! CHECK:  %[[VAL_2:.*]] = fir.convert %arg0 : (!fir.box<!fir.array<?x?xf32>>) -> !fir.box<none>
 subroutine lbound_test_2(a, dim, res)
   real, dimension(:, 2:) :: a
```

Comment:
It is weird that `arith.constant 2` is not showing up in  the new test, these tests should precisely  show that the lower bound constants get used somehow by the generated code.